src/libcharon/plugins/kernel_super/system_events/system_events.h

```diff
@@ -0,0 +1,28 @@
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
All the system_events files are not included from c++ files, remove the useless extern "C"